server/routes/utils/__tests__/savedSearchReportHelper.test.ts

```diff
@@ -909,7 +912,7 @@ describe('test create saved search report', () => {
           'geoip.location': { lon: -0.1, lat: 51.5 },
           customer_birth_date: '2023-04-26T04:34:32Z',
           order_date: '2023-04-26T04:34:32Z',
-          products: { created_on: '2023-04-26T04:34:32Z' },
+          products: [{ created_on: '2023-04-26T04:34:32Z' }],
```

Comment:
Let's add more objects and fields in this array for better testing the array iteration.